airgun/views/subscription.py

```diff
@@ -0,0 +1,185 @@
+        refresh_button = Button('Refresh')
+        delete_button = Button('Delete')
+
+    @View.nested
+    class manifest_history(SatTab):
+        TAB_NAME = "Manifest History"
+        table = SatTable(
+                locator='//div[@id="manifest-history-tabs"]//table',
+                column_widgets={
+                    'Status': Text(),
+                    'Message': Text(),
+                    'Timestamp': Text()
+                }
+        )
+
+    @property
+    def is_displayed(self):
+        return (self.browser.wait_for_element(
+                self.close_button, visible=True, exception=False) is not None
+                and 'in' in self.browser.classes(self))
+
+    def wait_animation_end(self):
+        wait_for(
+                lambda: 'in' in self.browser.classes(self),
+                handle_exception=True, logger=self.logger, timeout=10
+        )
+
+
+class DeleteManifestConfirmationView(BaseLoggedInView):
+    ROOT = ('//div[@role="dialog" and @tabindex]'
+            '[div//h4[text()="Confirm delete manifest"]]')
+    message = Text('.//div[@class="modal-body"]')
+    delete_button = Button('Delete')
+    cancel_button = Button('Cancel')
+
+    @property
+    def is_displayed(self):
+        return (self.browser.wait_for_element(
+                self.delete_button, visible=True, exception=False) is not None
+                and 'in' in self.browser.classes(self))
+
+    def wait_animation_end(self):
+        wait_for(
+                lambda: 'in' in self.browser.classes(self),
+                handle_exception=True, logger=self.logger, timeout=10
+        )
+
+
+class AddSubscriptionView(BaseLoggedInView):
+    breadcrumb = BreadCrumb()
+    table = SatTable(
+            locator='.//table',
+            column_widgets={
+                'Subscription Name': Text('.//a'),
+                'Quantity to Allocate': TextInput(locator='.//input'),
+            }
+    )
+    submit_button = Button('Submit')
+    cancel_button = Button('Cancel')
+
+    @property
+    def is_displayed(self):
+        return self.browser.wait_for_element(
+                self.table, visible=True, exception=False) is not None
+
+
+class SubscriptionDetailsView(BaseLoggedInView):
+    breadcrumb = BreadCrumb()
+
+    @View.nested
+    class details(SatTab):
+
+        @property
+        def provided_products(self):
+            return [elem.text
+                    for elem
+                    in self.browser.elements(
+                        ("//h2[text()='Provided Products']"
+                         "/following::ul/li"))]
+
+    @View.nested
+    class enabled_products(SatTab):
+        TAB_NAME = "Enabled Products"
+
+        @property
+        def enabled_products_list(self):
+            locator = ("//div[contains(@class, 'list-group')]"
+                       "//div[contains(@class, 'list-group-item-heading')]"
+                       )
+            try:
+                self.browser.wait_for_element(locator, visible=True)
```

Comment:
I feel something wrong on that logic. That level should not be ambiguous and raise exception in case element is not found. Otherwise you will wait here for 12 seconds for no reason